src/app/clusters/drlc-server/Commands.h

```diff
@@ -0,0 +1,1681 @@
+#if MTR_ENABLE_PROVISIONAL
+        params.loadControlProgramID = [NSData dataWithBytes:mRequest.loadControlProgramID.data() length:mRequest.loadControlProgramID.size()];
+#endif // MTR_ENABLE_PROVISIONAL
+        uint16_t repeatCount = mRepeatCount.ValueOr(1);
+        uint16_t __block responsesNeeded = repeatCount;
+        while (repeatCount--) {
+            [cluster unregisterLoadControlProgramRequestWithParams:params completion:
+                                                                              ^(NSError * _Nullable error) {
+                                                                                  responsesNeeded--;
+                                                                                  if (error != nil) {
+                                                                                      mError = error;
+                                                                                      LogNSError("Error", error);
+                                                                                      RemoteDataModelLogger::LogCommandErrorAsJSON(@(endpointId), @(clusterId), @(commandId), error);
+                                                                                  }
+                                                                                  if (responsesNeeded == 0) {
+                                                                                      SetCommandExitStatus(mError);
+                                                                                  }
+                                                                              }];
+        }
+        return CHIP_NO_ERROR;
+    }
+
+private:
+    chip::app::Clusters::DemandResponseLoadControl::Commands::UnregisterLoadControlProgramRequest::Type mRequest;
+};
+
+#endif // MTR_ENABLE_PROVISIONAL
+#if MTR_ENABLE_PROVISIONAL
+/*
+ * Command AddLoadControlEventRequest
+ */
+class DemandResponseLoadControlAddLoadControlEventRequest : public ClusterCommand {
+public:
+    DemandResponseLoadControlAddLoadControlEventRequest()
+        : ClusterCommand("add-load-control-event-request")
+        , mComplex_Event(&mRequest.event)
+    {
+#if MTR_ENABLE_PROVISIONAL
+        AddArgument("Event", &mComplex_Event);
+#endif // MTR_ENABLE_PROVISIONAL
+        ClusterCommand::AddArguments();
+    }
+
+    CHIP_ERROR SendCommand(MTRBaseDevice * device, chip::EndpointId endpointId) override
+    {
+        constexpr chip::ClusterId clusterId = chip::app::Clusters::DemandResponseLoadControl::Id;
+        constexpr chip::CommandId commandId = chip::app::Clusters::DemandResponseLoadControl::Commands::AddLoadControlEventRequest::Id;
+
+        ChipLogProgress(chipTool, "Sending cluster (0x%08" PRIX32 ") command (0x%08" PRIX32 ") on endpoint %u", clusterId, commandId, endpointId);
+
+        dispatch_queue_t callbackQueue = dispatch_queue_create("com.chip.command", DISPATCH_QUEUE_SERIAL);
```

Comment:
The placement of these files does not look right: there should be no darwin-specific code in src/app/clusters nor should there be generated code in src/app